test/test_packages.py

```diff
@@ -0,0 +1,170 @@
+    # 2020-03-08 09:56:08 [    INFO] Trying to import jupyterhub (test_packages.py:127)
+    # [...]
+
+"""
+
+import logging
+
+import pytest
+
+from helpers import CondaPackageHelper
+
+LOGGER = logging.getLogger(__name__)
+
+# Mapping between package and module name
+PACKAGE_MAPPING = {
+    # Python
+    "matplotlib-base": "matplotlib",
+    "beautifulsoup4": "bs4",
+    "scikit-learn": "sklearn",
+    "scikit-image": "skimage",
+    # R
+    "randomforest": "randomForest",
+    "rsqlite": "DBI",
+    "rcurl": "RCurl",
+    "rodbc": "RODBC",
+}
+
+# List of packages that cannot be tested in a standard way
+EXCLUDED_PACKAGES = [
+    "tini",
+    "python",
+    "hdf5",
+    "conda-forge::blas[build",
+    "protobuf",
+    "r-irkernel",
+    "unixodbc",
+    "spylon-kernel",
+]
+
+
+@pytest.fixture(scope="function")
+def package_helper(container):
+    """Return a package helper object that can be used to perform tests on installed packages"""
+    return CondaPackageHelper(container)
+
+
+@pytest.fixture(scope="function")
+def packages(package_helper):
+    """Return the list of specified packages (i.e. packages explicitely installed excluding dependecnies)"""
+    return package_helper.specified_packages()
+
+
+def package_map(package):
+    """Perform a mapping between the python package name and the name used for the import"""
+    _package = package
+    if _package in PACKAGE_MAPPING:
+        _package = PACKAGE_MAPPING.get(_package)
+    return _package
+
+
+def excluded_package_predicate(package):
+    """Return whether a package is excluded from the list (i.e. a package that cannot be tested with standard imports)"""
+    return package in EXCLUDED_PACKAGES
+
+
+def python_package_predicate(package):
+    """Predicate matching python packages"""
+    return not excluded_package_predicate(package) and not r_package_predicate(package)
+
+
+def r_package_predicate(package):
+    """Predicate matching R packages"""
+    return not excluded_package_predicate(package) and package.startswith("r-")
+
+
+def _check_import_package(package_helper, command):
+    """Generic function executing a command"""
+    LOGGER.debug(f"Trying to import a package with [{command}] ...")
+    rc = package_helper.running_container.exec_run(command)
+    return rc.exit_code
+
+
+def check_import_python_package(package_helper, package):
+    """Try to import a Python package from the command line"""
+    return _check_import_package(package_helper, ["python", "-c", f"import {package}"])
+
+
+def check_import_r_package(package_helper, package):
+    """Try to import a R package from the command line"""
+    return _check_import_package(
+        package_helper, ["R", "--slave", "-e", f"library({package})"]
+    )
+
+
+def _import_packages(package_helper, filtered_packages, check_function):
+    """Test if packages can be imported
+    
+    Note: using a list of packages instead of a fixture for the list of packages since pytest prevent to use multiple yields
```

Comment:
```suggestion
    Note: using a list of packages instead of a fixture for the list of packages since pytest prevents use of multiple yields
```

Reply:
A french guy who writes in English ...

Reply:
You make fewer typos than I do as a English-speaker. 🤣